cloudinit/net/netops/bsd_netops.py

```diff
@@ -21,11 +21,15 @@ def add_route(
         gateway: Optional[str] = None,
         source_address: Optional[str] = None
     ):
-        pass
+        subp.subp(["route", "add", route, "-interface", interface])
+        if gateway and gateway != "0.0.0.0":
+            subp.subp(
+                ["route", "change", route, gateway],
+            )
```

Comment:
We don't support source selection

and i had to make this addition in two steps, because when i did it in a single step, the result was nonsense.